pkg/akv/backend.go

```diff
@@ -0,0 +1,96 @@
+// Package akv implements backend for Azure Key Vault secrets
+package akv
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/keyvault/keyvault"
+	kvauth "github.com/Azure/azure-sdk-for-go/services/keyvault/auth"
+	"github.com/containersolutions/externalsecret-operator/pkg/backend"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var log = ctrl.Log.WithName("akv")
+
+// Backend is the needed structure to access the Azure Key Vault service
+type Backend struct {
+	client   keyvault.BaseClient
+	keyvault string
+}
+
+// NewBackend gives you a new akv.Backend
+func NewBackend() backend.Backend {
+	return &Backend{}
+}
+
+func init() {
+	backend.Register("akv", NewBackend)
+}
+
+// Init will initialize and authorize a local client responsible for accessing the Azure Key Vault service
+func (a *Backend) Init(parameters map[string]interface{}, credentials []byte) error {
+
+	akvCred := AzureCredentials{}
+	jsonErr := json.Unmarshal(credentials, &akvCred)
+	if jsonErr != nil {
+		log.Error(jsonErr, "")
+		return jsonErr
+	}
+
+	file, err := ioutil.TempFile("/tmp", "akv")
+	if err != nil {
+		log.Error(err, "")
+		return err
+	}
+
+	_, err = file.Write(credentials)
+	if err != nil {
+		log.Error(err, "error writing the temp file")
+		return err
+	}
+
+	//Remove this after/if this issue gets fixed: https://github.com/Azure/azure-sdk-for-go/issues/13641
```

Comment:
There was a merge reference in this issue, it might got solved 🤔 could u take a look there?

Reply:
Fixed it based on their change, but I also noticed it'll change again in the near future so I'll keep an eye on it.